src/components/UsersDetailsCard.jsx

```diff
@@ -33,10 +33,45 @@ const tableHeaders = [
   },
 ];
 
+function reducer(state, action) {
```

Comment:
no need to create a reducer function in memory. I think it is very small and we can directly pass it as an arguement   to useReducer hook.

for eg:
```
useReducer( (state, newState) => ( {...state, ...newState } ),
 {
    selectedArray: [],
    searchText: "",
    open: false,
    selectAll: false,
    buttonArray: [],
    selectedButton: "bt_1",
  });
  ```